twine/repository.py

```diff
@@ -90,7 +90,21 @@ def _make_adapter_with_retries() -> adapters.HTTPAdapter:
     def _make_user_agent_string() -> str:
         from twine import cli
 
-        dependencies = cli.list_dependencies_and_versions()
+        ignored_dependencies = {
+            "SecretStorage",
+            "bleach",
+            "colorama",
+            "keyring",
+            "readme-renderer",
+            "rich",
+            "tqdm",
+        }
```

Comment:
I would rather make this a list of versions to show, rather than ones to ignore. I might also do it in `cli.list_dependencies_and_versions`, effectively reverting https://github.com/pypa/twine/pull/858/commits/0983ab9c8e3c3627e0200ae697bf6c8bc83a4bcb.

Reply:
OK, here you go.